elastic/kibana
# [Synthetics] use_monitor_name - remove public saved object client usage

## Summary

Relates to https://github.com/elastic/kibana/issues/153399

Addresses the following task from the referenced issue
- [ ] https://github.com/elastic/kibana/blob/main/x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/hooks/use_monitor_name.ts#L38

Adjusts the `use_monitor_name` hook to remove references to the public saved object client.

This also consolidates two hooks named `use_monitor_name` into 1.

It also moves a few hooks relates to this change from specific component folders into the common public hooks folder.

### Testing

This PR touches two places: the UI validation preventing monitors from having the same name and the monitor selector on the monitor details page

1. Create a monitor
2. Attempt to create a second monitor with the same name. A validation error should appear
3. Change the name. The validation error should disappear. Save the monitor
4. Navigate to the monitor details page
5. Attempt to use the monitor selector (next to the monitor name). Test against regression.
